sector-builder/src/scheduler.rs

```diff
@@ -386,25 +344,77 @@ impl<T: KeyValueStore, S: SectorStore> SectorMetadataManager<T, S> {
     pub fn handle_seal_result(
         &mut self,
         sector_id: SectorId,
-        result: Result<SealedSectorMetadata>,
+        sector_access: String,
+        sector_path: PathBuf,
+        result: Result<SealOutput>,
     ) {
         // scope exists to end the mutable borrow of self so that we can
         // checkpoint
         {
             let staged_state = &mut self.state.staged;
             let sealed_state = &mut self.state.sealed;
 
-            match result {
-                Err(err) => {
-                    if let Some(staged_sector) = staged_state.sectors.get_mut(&sector_id) {
-                        staged_sector.seal_status =
-                            SealStatus::Failed(format!("{}", err_unrecov(err)));
+            let staged_sector = staged_state
+                .sectors
+                .get_mut(&sector_id)
+                .expect("missing staged sector");
+
+            let _ = result
```

Comment:
why are you throwing away the result?

Reply:
I can't do anything with the `result` at this point. I've already run all side effects via lambdas passed to `map_err` and `map`.

Reply:
I refactored such that I didn't have to throw away the result :)